packages/plexus/src/LayeredDigraph/SvgLayer.tsx

```diff
@@ -20,40 +20,69 @@ import { assignMergeCss, getProps } from './utils';
 import TNonEmptyArray from '../types/TNonEmptyArray';
 import ZoomManager from '../ZoomManager';
 
-type TProps<T = {}, U = {}> = TSetOnContainer<T, U> & {
-  classNamePrefix?: string;
-  defs?: TNonEmptyArray<TDefEntry<T, U>>;
-  graphState: TExposedGraphState<T, U>;
-};
+type TProps<T = {}, U = {}> = Record<string, unknown> &
+  TSetOnContainer<T, U> & {
+    classNamePart: string;
+    getClassName: (name: string) => string;
+    defs?: TNonEmptyArray<TDefEntry<T, U>>;
+    extraWrapper?: Record<string, unknown>;
+    graphState: TExposedGraphState<T, U>;
+    standalone?: boolean;
+    topLayer?: boolean;
+  };
 
-export default class SvgDoc<T = {}, U = {}> extends React.PureComponent<TProps<T, U>> {
+const STYLE: React.CSSProperties = { minHeight: '100%', minWidth: '100%', position: 'absolute' };
+
+export default class SvgLayer<T = {}, U = {}> extends React.PureComponent<TProps<T, U>> {
   render() {
-    const { children, classNamePrefix, defs, graphState, setOnContainer } = this.props;
-    const { zoomTransform } = graphState;
+    const {
+      children,
+      classNamePart,
+      getClassName,
+      defs,
+      extraWrapper,
+      graphState,
+      setOnContainer,
+      standalone,
+      topLayer,
+    } = this.props;
+
     const containerProps = assignMergeCss(getProps(setOnContainer, graphState), {
-      className: `${classNamePrefix} ${classNamePrefix}-LayeredDigraph--SvgDoc`,
-      style: { minHeight: '100%', minWidth: '100%' },
+      className: getClassName(classNamePart),
     });
-
-    return (
-      <svg {...containerProps}>
+    let content = (
+      <g {...containerProps}>
         {defs && (
           <defs>
             {defs.map(defEntry => (
               <SvgDefEntry<T, U>
                 key={defEntry.localId}
                 {...defEntry}
-                classNamePrefix={classNamePrefix}
+                getClassName={getClassName}
                 graphState={graphState}
               />
             ))}
           </defs>
         )}
+        {children}
+      </g>
+    );
+    if (extraWrapper) {
+      content = <g {...extraWrapper}>{content}</g>;
```

Comment:
since the consumer can add props to the "standard" `g`, what is the use case for the extra `g`?

Reply:
Ultimately, this results in:
```
<svg>
  <g className={ consumerDefinedClassName } style={ CONSTANT_STYLE }>
    <g className={ anotherConsumerDefinedClassName } transform={ basedOnZoomManager }>
      <g { ...extraWrapperPropsDefinedByConsumer }>
        <g { ...containerPropsAlsoDefinedByConsumer }>
          {defs /* ... */}
          {children}
        </g>
      </g>
    </g>
  </g>
</svg>
```
I don't know if there is a downside to having multiple nested `<g>`s, but it seems like this is more complicated than necessary. It leaves me wondering when a consumer would define something through `setOnContainer` vs `extraWrapper`, and which of the two `classNames` that I defined I should style.

Reply:
`extraWrapper` is for internal use only; it's not part of the public API.. So, the public API is not ambiguous, the user would use `setOnContainer`. 

Mainly, `extraWrapper` so the edges can have the `stroke` defaulted to `#000` without it inadvertently overriding the CSS styling set through the public API. E.g. the container the user applies `setOnContainer` to inherits the `stroke="#000"` but can override it via CSS styling. 

Right now there is a `<g>` for each thing that needs to be accomplished. To reduce the nesting would entail combining some `<g>` elements, which will be a bit more complex to map the code to the DOM elements.

I agree the levels of nesting could be reduced. But I'm pretty sure it's still very light as this happens on the scale of the number of layers not the number of vertices or edges in the graph. I would think the perf impact is negligible.